### ISSUE TITLE
Respect update.serial setting of BOSHDeployment

### ISSUE THREAD

**Issue by kramerul** [cloudfoundry-incubator/cf-operator#636]
Currently all Instance Groups are starting in parallel. For [scf](https://github.com/SUSE/scf), this will produce a lot of container crashes (e.g. api needs uaa running). This increases the boot times, because every crash doubles the time until the next restart takes place.

The boot time could be reduced by respecting the `update.serial` setting in the Manifest.
We did some experiments with  [scf](https://github.com/SUSE/scf)

|          | Serial Boot   | Parallel Boot |
|--------|----------------|---------------|
| Time |  Between 14 and 16 min | Between 20 and 30 minutes|
| Crashes | No crashes               |  Many crashes |

We implemented this with the help of an additional init container, which was running before `boshPreStartInitContainers` and waits until `nslookup` of the headless service of the previous instance group was not failing. (For details have a look at our [branch based on 0.4.0 ](https://github.com/akhinos/cf-operator/tree/poc-init-order))

**Comment by viovanov** [comment]
Thanks @kramerul - super useful!

**Comment by c0d1ngm0nk3y** [comment]
As a POC, we assumed `update.serial` always to be `true`. We have added an `initContainer` to each `pod` which waits for the headless service of the predecessor. For now, we hardcoded the dependencies instead of reading it from the manifest to see if it is worth the effort.

There are 2 effect that makes it worth the effort imho:

1) The start time is faster and more reliable (no `CrashLoopBackOff`)
2) The state of the cluster is much cleaner during startup - no crashes, only `Init` or `Running`
